feat(nexus): adding additional statistics for rebuild

dsavitskiy opened the pull request.

dsavitskiy commented:
bors try

dsavitskiy commented:
bors try

dsharma-dc commented:
These changes appear to be on top of PR #1293. Are you planning to retain 1293 or, go ahead with all changes on this PR and drop 1293?

dsavitskiy commented:
@dsharma-dc Normally I would prefer to avoid such situation. But when I need to have a PR on top of another, I first wait until I merge the base PR first, and then I rebase the dependent one and the base commits are gone.
Reviews are supposed to review only the relevant commit in this PR. The base commits are reviewed in a separate PR.

dsavitskiy commented:
bors try

dsavitskiy committed:
feat(nexus): adding additional statistics for rebuild

Blocks actually tranferred and blocks remaining to transfer added.
Rebuild history record code cleaned up.
Rebuild history removed from v0 API.

Signed-off-by: Dmitry Savitskiy <dmitry.savitskiy@datacore.com>

dsavitskiy commented:
bors try

tiagolobocastro commented on io-engine/src/rebuild/rebuild_job_backend.rs:
```diff
@@ -367,33 +370,39 @@ impl RebuildJobBackend {
+        };
+
+        info!(
+            "State: {state}, Nexus: {nex}, Src: {src}, Dst: {dst}: {res:?}",
```
Previously progress was shown, should we keep it as it's kind useful to see it?

dsavitskiy replied:
It is still shown via {res:?}, res being the stats. Debug output for Stats includes progress.

tiagolobocastro replied:
doh, sorry I missed that

dsavitskiy commented:
bors merge

chriswldenyer reviewed (approved):
Looks reasonable

tiagolobocastro commented:
bors merge
